app/views/articles/show.html.erb

```diff
@@ -95,7 +95,9 @@
       <% if !@article.published %>
         <div class="crayons-notice crayons-notice--danger mb-4">
           <strong>Unpublished Post.</strong> This URL is public but secret, so share at your own discretion.
-          <a href="<%= @article.path %>/edit" class="fw-bold">Click to edit</a>.
+          <% if user_signed_in? && @article.user_id == current_user.id %>
```

Comment:
Maybe we can use the `ArticlePolicy.update?` here? 

https://github.com/forem/forem/blob/f29f4b7840a5ca1f56620b31e89e6d18d6ee65ac/app/policies/article_policy.rb#L2-L4

you can check the policy in the view with something like:

```erb


<% if policy(@post).update? %>
  <%= link_to "Edit post", edit_post_path(@post) %>
<% end %>
```

taken from https://github.com/varvet/pundit#policies

this way we centralize the intent